core/consensus/grandpa/impl/grandpa_impl.cpp

```diff
@@ -477,19 +473,60 @@ namespace kagome::consensus::grandpa {
       return;
     }
 
-    std::shared_ptr<VotingRound> target_round = selectRound(fin.round_number);
-    if (not target_round) {
-      if (current_round_->roundNumber() < fin.round_number) {
-        if (catch_up_request_suppressed_until_ > clock_->now()) {
-          catch_up_request_suppressed_until_ =
-              clock_->now() + catch_up_request_suppression_duration_;
-          current_round_->doCatchUpRequest(peer_id);
-        }
-      }
+    if (fin.justification.round_number != fin.round_number) {
+      logger_->warn("Fin message is inconsistent by round");
+      return;
+    }
+
+    if (fin.justification.block_info != fin.vote) {
+      logger_->warn("Fin message is inconsistent by block");
```

Comment:
To be honest these two messages are not very clear for me.

Reply:
Ok, I'll change it